src/Common/src/Interop/Windows/winhttp/Interop.SafeInternetHandle.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class WinHttp
+    {
+        // TODO: Move SafeHandleZeroOrMinusOneIsInvalid class to Common
+        internal abstract class SafeHandleZeroOrMinusOneIsInvalid : SafeHandle
+        {
+            protected SafeHandleZeroOrMinusOneIsInvalid(bool ownsHandle)
+                : base(IntPtr.Zero, ownsHandle)
+            {
+            }
+
+            public override bool IsInvalid
+            {
+                get { return this.handle == IntPtr.Zero || this.handle == (IntPtr)(-1); }
+            }
+        }
+
+        internal class SafeInternetHandle : SafeHandleZeroOrMinusOneIsInvalid
```

Comment:
The rest of the SafeHandles are in Common\src\Microsoft\Win32\SafeHandles as TypeName.Platform.cs.  So SafeInternetHandle.Windows.cs

We seem to have done away with SafeHandleZeroIsInvalid and SafeHandleZeroOrMinusOneIsInvalid, and just unrolled the logic into the leaf types.